lib/portlayer/storage/image_cache.go

```diff
@@ -300,3 +298,54 @@ func (c *NameLookupCache) ListImages(ctx context.Context, store *url.URL, IDs []
 
 	return imageList, nil
 }
+
+// DeleteImage deletes an image from the image store.  If it is in use or is being inheritted from, then this will return an error.
+func (c *NameLookupCache) DeleteImage(ctx context.Context, image *Image) error {
+	infof("DeleteImage: deleting %s", image.Self())
+
+	// Check the image exists.  This will rehydrate the cache if necessary.
+	_, err := c.GetImage(ctx, image.Store, image.ID)
+	if err != nil {
+		return errorf("DeleteImage: %s", err)
+	}
+
+	// get the relevant cache
+	c.storeCacheLock.Lock()
+	indx := c.storeCache[*image.Store]
+	c.storeCacheLock.Unlock()
+
+	hasChildren, err := indx.HasChildren(image.Self())
+	if err != nil {
+		return errorf("DeleteImage: %s", err)
```

Comment:
in use -  will need to define that error.  Will update this change with a new error struct for that use case if it doesnt already exist in the `io` package.

not found - will use the `io` package's variant.

permission denied - we don't support mode bits or uid/gid of vmdk images.  

internal error - aren't all undefined errors internal errors?